concurrent_test.go

```diff
@@ -292,138 +236,212 @@ func runWorker(t *testing.T, w worker, errCh chan error) (historyRecords, error)
 	return rs, err
 }
 
-type worker interface {
-	name() string
-	run() (historyRecords, error)
-}
-
-type readWorker struct {
+type worker struct {
 	id int
 	db *btesting.DB
 
 	bucket []byte
 	keys   []string
 
-	readTime duration
+	conf concurrentConfig
 
 	errCh  chan error
 	stopCh chan struct{}
 
 	t *testing.T
 }
 
-func (r *readWorker) name() string {
-	return fmt.Sprintf("readWorker-%d", r.id)
+func (w *worker) name() string {
+	return fmt.Sprintf("worker-%d", w.id)
 }
 
-func (r *readWorker) run() (historyRecords, error) {
+func (w *worker) run() (historyRecords, error) {
 	var rs historyRecords
 	for {
 		select {
-		case <-r.stopCh:
-			r.t.Log("Reading transaction finished.")
+		case <-w.stopCh:
+			w.t.Logf("Worker %q finished.", w.name())
```

Comment:
`Worker "worker-0" finished.` sounds repetitive. Do we really need `name()`?

Reply:
Yes, but seems not a big deal. Updated anyway.